### PR TITLE
Added new metric, ratio of comment lines to code lines 2

### PR DESCRIPTION
**Pull request by eolivi**
continuing the work done in https://github.com/metrixplusplus/metrixplusplus/pull/31 by @holyraspi

https://github.com/metrixplusplus/metrixplusplus/pull/31

### CODE REVIEW

**Comment by avkonst** [comment]
hi, thanks for contribution. I hope you verified that the metrics reporting is correct in your code and in the samples?

**Comment by avkonst** [comment]
could you please verify test failures? probably gold files need to be updated
also could you please update the table on website with the lit of available metrics?

**Comment by avkonst** [comment]
Thanks for clarification. The correct definition is the second one:

CR = lines_comments / (lines_codes + lines_comments)
It means no single line of comment would give CR=0. Only comments and no lines of code would give CR=1. It means the range for CR is from 0 to 1. Which is consistent with the word 'ratio' or 'percentage' of comments.

If you go with the first definition, CR can jump to very high number.

**Comment by eolivi** [comment]
Hi,
> Which is consistent with the word 'ratio' or 'percentage' of comments.

I do not fully agree (one could have a 3:1 ratio...), but never mind, I reverted to the second option.

One issue I see while running the test using python2 and python3, is that the output is slightly different. That could lead to wrong matches with the gold files...

**Comment by avkonst** [comment]
"one could have a 3:1 ratio..."
- what would be the handling when it is 3:0?
- if you need the first definition, it can probably be done with a command line option, if you wish... but the handling for 0 code lines should be defined somehow...

**Comment by eolivi** [comment]
Some tests are failing, as I said, I believe it is due to different output using python 2 vs 3.
So I don't know what to do ... please have a look

**Comment by avkonst** [comment]
Could you truncate std.code.ratio:comments results to 3 significant digits after comma? I think it will fix the problem. All fractional metrics had this problem. I am sorry I can not have a look at it right now as I am too busy with other projects, but you almost done it, just a tiny bit left. when you save your new metric, make sure it is only 3 significant digits after comma.

File: metrixpp/ext/std/tools/view.py [hunk 1]
```diff
@@ -544,7 +544,7 @@ def cout_txt_regions(path, regions, indent = 0):
                 if field == '__diff__':
                     continue
                 if field in list(diff_data.keys()):
-                    diff_str = " [" + ("+" if diff_data[field] >= 0 else "") + str(diff_data[field]) + "]"
+                    diff_str = " [" + ("+" if diff_data[field] >= 0 else "") + "{0:.3g}".format(diff_data[field]) + "]"
```
Comment (eolivi) [inline metrixpp/ext/std/tools/view.py:547]: I had to change this in order to get the same output using python2 or python3. 

For example, the problem using str was that it was printing 0.07500000000000001 using python2 and 0.075 using python3.

[ I am not sure it is the right to solve this pb, and/or if it should be used elsewhere too. ]

**Comment by eolivi** [comment]
bump please @avkonst ?
What's wrong now ?

**Comment by avkonst** [comment]
nothing is wrong, just missed it... you may also want to consider to add you metric to website docs..

**Pull request merged**